lib/ansible/modules/network/cloudengine/ce_ntp_auth.py

```diff
@@ -0,0 +1,588 @@
+        <isReliable></isReliable>
+      </ntpAuthKeyCfg>
+    </ntpAuthKeyCfgs>
+  </ntp>
+</filter>
+"""
+
+CE_NC_GET_NTP_AUTH_ENABLE = """
+<filter type="subtree">
+  <ntp xmlns="http://www.huawei.com/netconf/vrp" content-version="1.0" format-version="1.0">
+    <ntpSystemCfg>
+      <isAuthEnable></isAuthEnable>
+    </ntpSystemCfg>
+  </ntp>
+</filter>
+"""
+
+CE_NC_MERGE_NTP_AUTH_CONFIG = """
+<config>
+  <ntp xmlns="http://www.huawei.com/netconf/vrp" content-version="1.0" format-version="1.0">
+    <ntpAuthKeyCfgs>
+      <ntpAuthKeyCfg operation="merge">
+        <keyId>%s</keyId>
+        <mode>%s</mode>
+        <keyVal>%s</keyVal>
+        <isReliable>%s</isReliable>
+      </ntpAuthKeyCfg>
+    </ntpAuthKeyCfgs>
+  </ntp>
+</config>
+"""
+
+CE_NC_MERGE_NTP_AUTH_ENABLE = """
+<config>
+  <ntp xmlns="http://www.huawei.com/netconf/vrp" content-version="1.0" format-version="1.0">
+    <ntpSystemCfg operation="merge">
+      <isAuthEnable>%s</isAuthEnable>
+    </ntpSystemCfg>
+  </ntp>
+</config>
+"""
+
+CE_NC_DELETE_NTP_AUTH_CONFIG = """
+<config>
+  <ntp xmlns="http://www.huawei.com/netconf/vrp" content-version="1.0" format-version="1.0">
+    <ntpAuthKeyCfgs>
+      <ntpAuthKeyCfg operation="delete">
+        <keyId>%s</keyId>
+      </ntpAuthKeyCfg>
+    </ntpAuthKeyCfgs>
+  </ntp>
+</config>
+"""
+
+
+class NtpAuth(object):
+    """Manage ntp authentication"""
+
+    def __init__(self, argument_spec):
+        self.spec = argument_spec
+        self.module = None
+        self.init_module()
+
+        # ntp_auth configration info
+        self.key_id = self.module.params['key_id']
+        self.password = self.module.params['auth_pwd'] or None
+        self.auth_mode = self.module.params['auth_mode'] or None
+        self.auth_type = self.module.params['auth_type']
+        self.trusted_key = self.module.params['trusted_key']
+        self.authentication = self.module.params['authentication'] or None
+        self.state = self.module.params['state']
+        self.check_params()
+
+        self.ntp_auth_conf = dict()
+        self.key_id_exist = False
+        self.cur_trusted_key = 'disable'
+
+        # state
+        self.changed = False
+        self.updates_cmd = list()
+        self.results = dict()
+        self.proposed = dict()
+        self.existing = list()
+        self.end_state = list()
+
+        self.get_ntp_auth_exist_config()
+
+    def check_params(self):
+        """Check all input params"""
+
+        if not self.key_id.isdigit():
+            self.module.fail_json(
+                msg='Error: key_id is not digit.')
+
+        if (long(self.key_id) < 1) or (long(self.key_id) > 4294967295):
+            self.module.fail_json(
+                msg='Error: The length of key_id is between 1 and 4294967295.')
+
+        if self.state == "present":
+            if (not self.password) or (not self.auth_mode):
+                self.module.fail_json(
+                    msg='Error: Please input password and auth_mode.')
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
required_together
```